process.go

```diff
@@ -233,31 +224,61 @@ func runTestCase(ts *TestSuite, tc *TestCase, l *log.Entry, detailsLevel string)
 	l.Infof("start")
 	for _, step := range tc.TestSteps {
 
-		t, err := getExecutor(step)
+		e, err := getExecutorWrap(step)
 		if err != nil {
 			tc.Errors = append(tc.Errors, Failure{Value: err.Error()})
 			break
 		}
 
-		result, err := t.Run(l, aliases, step)
+		runTestStep(e, tc, step, l, detailsLevel)
+
+		if detailsLevel != DetailsLow {
+			bars[ts.Package].Increment()
+		}
+		if len(tc.Failures) > 0 {
+			break
+		}
+	}
+	l.Infof("end")
+}
+
+func runTestStep(e *executorWrap, tc *TestCase, step TestStep, l *log.Entry, detailsLevel string) {
+
+	var isOK bool
+	var errors []Failure
+	var failures []Failure
+
+	for retry := 0; retry <= e.retry; retry++ {
```

Comment:
it's <=, not <, otherwise, with retry 0 (default value) -> nothing will be runned